### PR TITLE
337 subscription id field

### PR DESCRIPTION
**Pull request by DutchBen**
Subscription field that is used to select a subscription based on some filters
The field suplplies a modal with a tab for favorites in V1 but this modal is dropped and wil not return in V2

### CODE REVIEW

File: packages/orchestrator-ui-components/src/components/WfoForms/formFields/SubscriptionField.tsx [hunk 1]
```diff
@@ -0,0 +1,380 @@
...
+} from 'uniforms';
+
+import { useOrchestratorTheme } from '../../../hooks';
+import { FieldProps, Option, PortMode, ProductTag } from './types';
+import { getPortMode } from './utils';
+import { useGetSubscriptionDropdownOptions } from '../../../hooks/surf/useGetSubscriptionDropdownOptions';
+import { SubscriptionDropdownOption } from '../../../types';
+import { getReactSelectInnerComponentStyles } from './reactSelectStyles';
+
+const subscriptionFieldStyling = css`
```
Comment (ricardovdheijden) [inline packages/orchestrator-ui-components/src/components/WfoForms/formFields/SubscriptionField.tsx:42]: Try to have the styling out of the component

**Review by ricardovdheijden — APPROVED** [review]

File: packages/orchestrator-ui-components/src/components/WfoForms/formFields/SubscriptionField.tsx [hunk 2]
```diff
@@ -0,0 +1,380 @@
...
+        (option: Option) => option.value === value,
+    );
+
+    const isDisabled = disabled || readOnly || isFetching;
+
+    return (
+        <EuiFlexItem css={subscriptionFieldStyling} grow={1}>
+            <section
+                {...filterDOMProps(props)}
+                className={`${className} subscription-field${
```
Comment (acidjunk) [inline packages/orchestrator-ui-components/src/components/WfoForms/formFields/SubscriptionField.tsx:319]: You can see how I fixed the styling of acceptField. Probably the same pattern can be applied here.

File: packages/orchestrator-ui-components/src/components/WfoForms/formFields/types.ts [hunk 3]
```diff
@@ -53,3 +53,19 @@ export interface ServicePort {
     nonremovable?: boolean;
     modifiable?: boolean;
 }
+
```
Comment (acidjunk) [inline packages/orchestrator-ui-components/src/components/WfoForms/formFields/types.ts:56]: Can this be moved to a surf folder? They are very SURF specific.

File: packages/orchestrator-ui-components/src/messages/en-US.json [hunk 4]
```diff
@@ -70,6 +70,10 @@
                 "untaggedPortInUse": "This service port is already in use and cannot be chosen",
                 "taggedOnly": "VLAN is only relevant for SN7 MSP or SN8 SP in tagged mode, not for link_member or untagged ports.",
                 "loadingIms": "Loading VLAN info from IMS"
+            },
+            "subscription": {
```
Comment (acidjunk) [inline packages/orchestrator-ui-components/src/messages/en-US.json:74]: Backend? (probably not, so ignore.)

The `loadingIms` is also very SURF specific. I know that's not your doing.

File: packages/orchestrator-ui-components/src/types/forms.ts [hunk 5]
```diff
@@ -37,3 +37,22 @@ export type FieldProps<
         description?: string;
     } & Extra
 >;
+
+export interface FavoriteSubscriptionStorage {
+    subscription_id: string;
+    customName: string;
+}
+
+export interface ServicePortFilterItem {
```
Comment (acidjunk) [inline packages/orchestrator-ui-components/src/types/forms.ts:46]: SURF specific

**Review by acidjunk — APPROVED** [review]

File: packages/orchestrator-ui-components/src/components/WfoForms/formFields/SubscriptionField.tsx (hunk 1 above)
Comment (DutchBen) [reply packages/orchestrator-ui-components/src/components/WfoForms/formFields/SubscriptionField.tsx:42]: Moved it back to its own file

File: packages/orchestrator-ui-components/src/components/WfoForms/formFields/types.ts (hunk 3 above)
Comment (DutchBen) [reply packages/orchestrator-ui-components/src/components/WfoForms/formFields/types.ts:56]: Create a surf folder in the form directoryt and moved them there

File: packages/orchestrator-ui-components/src/messages/en-US.json (hunk 4 above)
Comment (DutchBen) [reply packages/orchestrator-ui-components/src/messages/en-US.json:74]: I propose to fix this later when we are separating the surf app

**Pull request merged**